src/MOI_wrapper/MOI_wrapper.jl

```diff
@@ -95,6 +97,10 @@ mutable struct Optimizer <: MOI.AbstractOptimizer
     Optimizer() = new(nothing, nothing, _Solution(), false, Dict{Symbol,Any}())
 end
 
+function MOI.default_cache(::Optimizer, ::Type{pfloat})
```

Comment:
So how about adding an option to the optimizer constructor for this? 
```julia
ECOS.Optimizer(; use_optimizer_cache::Bool = false)
```
Then you could have:
```Julia
function MOI.default_cache(model::Optimizer, ::Type{pfloat})
    if model.use_optimizer_cache
        return MOI.Utilities.UniversalFallback(OptimizerCache())
    end
    return MOI.Utilities.UniversalFallback(MOI.Utilities.Model{pfloat}())
end
```

This lets us do it without breaking, and avoids any unforeseen MOI issues.

Reply:
This should be the default. We want users to have good performance by default. We shouldn't tag v1.0 for solvers that use `MatrixOfConstraint` before we figure out if we want to use `default_cache` or not.
For ECOS, I think tagging v1.1 is fine.

Reply:
The number of PRs fixing things in MOI is one reason why I was hesitant. We haven't battle tested MatrixOfConstraints or `default_cache`, so it seems sensible to leave better performance as an opt-in and not risk breaking things.

Reply:
I'd rather have an option to `instantiate` than to every solver constructor.
If there are bugs, I'd rather make them apparent and fix them rather than try to hide them.
Over the 4 PRs, two are not related to MatrixOfConstraints, it's bridges and the missing canonicalize was quite bad, it's good we found it. The fourth one is not really a bug, it's more for convenience.